src/App.jsx

```diff
@@ -3,41 +3,92 @@ import { RouterProvider, createBrowserRouter } from "react-router-dom";
 import api from "../api";
 import Homepage from "./pages/home-page";
 import ErrorPage from "./pages/error-page";
+import ArticlePage from "./pages/article-page";
+import Layout from "./components/layout";
 import { CircularProgress } from "@mui/material";
 
-async function fetchArticles() {
-  const { data } = await api.getArticles;
-  return data;
-}
+function App() {
+  const [articles, setArticles] = useState([]);
+  const [votes, setVotes] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchArticles = async () => {
+    try {
+      const { data } = await api.getArticles();
+      setArticles(data);
+      const initialVotes = data.reduce((acc, article) => {
+        acc[article.article_id] = article.votes;
+        return acc;
+      }, {});
+      setVotes(initialVotes);
+    } catch (error) {
+      console.error("Failed to fetch articles", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchArticles();
+  }, [fetchArticles]);
```

Comment:
better to have regular states or variables as the dependencies of the useEffect hook rather than a function. Use function as a dependency may cause some future issues, not recommended